Description: logs to FIFO bugfixes
Author: Emanuele Rocca <ema@wikimedia.org>
Last-Update: 2018-11-22
Bug: https://github.com/apache/trafficserver/issues/4635
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/proxy/logging/LogFile.cc b/proxy/logging/LogFile.cc
index 7d70bbdde..7f1fa827b 100644
--- a/proxy/logging/LogFile.cc
+++ b/proxy/logging/LogFile.cc
@@ -166,13 +166,13 @@ LogFile::open_file()
 
   if (m_file_format == LOG_FILE_PIPE) {
     // setup pipe
-    if (mkfifo(m_name, S_IRUSR | S_IWUSR | S_IRGRP) < 0) {
-      if (errno != EEXIST) {
+    if (!file_exists) {
+      if (mkfifo(m_name, S_IRUSR | S_IWUSR | S_IRGRP) < 0) {
         Error("Could not create named pipe %s for logging: %s", m_name, strerror(errno));
         return LOG_FILE_COULD_NOT_CREATE_PIPE;
+      } else {
+        Debug("log-file", "Created named pipe %s for logging", m_name);
       }
-    } else {
-      Debug("log-file", "Created named pipe %s for logging", m_name);
     }
 
     // now open the pipe
diff --git a/proxy/logging/LogObject.cc b/proxy/logging/LogObject.cc
index c745bddb2..86597ebb4 100644
--- a/proxy/logging/LogObject.cc
+++ b/proxy/logging/LogObject.cc
@@ -1009,7 +1009,6 @@ LogObjectManager::_solve_filename_conflicts(LogObject *log_object, int maxConfli
             roll_file = false;
           } else {
             if (S_ISFIFO(s.st_mode)) {
-              unlink(filename);
               roll_file = false;
             }
           }
